plugins/dicom_viewer/server/__init__.py

```diff
@@ -206,19 +207,24 @@ def dicomSubstringSearchHandler(query, types, user=None, level=None, limit=0, of
     """
     if types != ['item']:
         raise RestException('The dicom search is only able to search in Item.')
+    if not isinstance(query, six.string_types):
```

Comment:
This just ensures that a non-string `query` (like a list or dict) doesn't slip in. [`six.string_types` is just a way to refer to strings](https://pythonhosted.org/six/#six.string_types) in both Python 2 and Python 3.